magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -459,11 +474,26 @@ object STS extends AWS {
   }
 
   def getAccountNumber(stsClient: StsClient): String = {
-    val callerIdentityResponse = stsClient.getCallerIdentity(GetCallerIdentityRequest.builder().build())
+    val callerIdentityResponse = stsClient.getCallerIdentity()
     callerIdentityResponse.account
   }
 }
 
+object SSM extends AWS {
+  def withSsmClient[T](keyRing: KeyRing, region: Region)(block: SsmClient => T): T = {
+    withResource(SsmClient.builder()
+      .credentialsProvider(provider(keyRing))
+      .overrideConfiguration(clientConfiguration)
+      .region(region.awsRegion)
+      .build())(block)
+  }
+
+  def getParameter(ssmClient: SsmClient, key: String): String = {
```

Comment:
the two `String` have the same type but actually mean something different.
when I see the same type twice for different purposes in the same function signature, I try wrapping in a value class so the purpose is still clear from the type.
This would make the withSsmClient on line 88 clearer especially.
However there is also a syntactic overhead of wrapping and unwrapping for pay for the extra clarity elsewhere.
```suggestion
  case class SsmKey(value: String) extends AnyVal
  case class SsmValue(value: String) extends AnyVal

  def getParameter(ssmClient: SsmClient, key: SsmKey): SsmValue = {
```

Reply:
We do that in some places but it's not consistent across the codebase so I'm inclined to leave this as is at the moment.